implementations/rust/ockam/ockam_command/src/node/list.rs

```diff
@@ -100,13 +102,22 @@ async fn run_impl(
         ));
     }
 
-    let list = opts
+    let plain = opts
         .terminal
         .build_list(&nodes, "Nodes", "No nodes found on this system.")?;
-    opts.terminal.stdout().plain(list).write_line()?;
+
+    let json = serde_json::to_string_pretty(&nodes).into_diagnostic()?;
+
+    opts.terminal
+        .stdout()
+        .plain(plain)
+        .json(json)
```

Comment:
@adrianbenavides thanks for the really valuable input. It's going to really help me with my future contributions.

Lastly, do you think it's worth it for me to check the `OutputFormat` to allocate only either the plain or JSON string? It would prevent us from having to allocate both the plain and json string when only one is needed.

To do that I could either get the `OutputFormat` from `opts` -> `CommandGlobalOpts` -> `GlobalArgs` or add `pub fn output_format(&self) -> &OutputFormat` to `Terminal`.

Let me know if you think that's necessary, otherwise we can just roll with this. Thanks!

Reply:
Nah, don't worry about that. In this layer (`ockam_command`) it's more important to have a clean and simple code rather than an optimized and complex solution. It's true that this part could be improved to avoid serializing the output in different formats, but that should be hidden from the commands' code, but it's out of the scope of this PR.